polaris/hub/client.py

```diff
@@ -326,13 +344,16 @@ def list_datasets(self, limit: int = 100, offset: int = 0) -> list[str]:
         dataset_list = [bm["artifactId"] for bm in response["data"]]
         return dataset_list
 
-    def get_dataset(self, owner: Union[str, HubOwner], name: str, verify_checksum: bool = True) -> Dataset:
+    def get_dataset(
+        self, owner: Union[str, HubOwner], name: str, verify_checksum: Optional[bool] = None
+    ) -> Dataset:
         """Load a dataset from the Polaris Hub.
 
         Args:
             owner: The owner of the dataset. Can be either a user or organization from the Polaris Hub.
             name: The name of the dataset.
-            verify_checksum: Whether to use the checksum to verify the integrity of the dataset.
+            verify_checksum: Whether to use the checksum to verify the integrity of the dataset. If None,
+                will infer a practical default based on the dataset's storage location.
```

Comment:
**Comment**
This feels like overloading a boolean arg to have more than two possible values. Also, it makes it unclear to me what "a practical default" would be. If I don't specify anything, will the checksum be verified?

Maybe something like an `Enum` or a `Literal` type with clearer options on what will happen?